README.md

```diff
@@ -1 +1,132 @@
-# jsonld-document-loader
\ No newline at end of file
+# JSON-LD Document Loader _(jsonld-document-loader)_
+
+[![Build Status](https://img.shields.io/github/workflow/status/digitalbazaar/jsonld-document-loader/Node.js%20CI)](https://github.com/digitalbazaar/jsonld-document-loader/actions?query=workflow%3A%22Node.js+CI%22)
+[![NPM Version](https://img.shields.io/npm/v/jsonld-document-loader.svg)](https://npm.im/@digitalbazaar/jsonld-document-loader)
+
+> A document loader API for jsonld.js.
+
+## Table of Contents
+
+- [Background](#background)
+- [Security](#security)
+- [Install](#install)
+- [Usage](#usage)
+- [Contribute](#contribute)
+- [Commercial Support](#commercial-support)
+- [License](#license)
+
+## Background
+
+TBD
+
+## Security
+
+TBD
+
+## Install
+
+- Node.js 12+ is required.
+
+### NPM
+
+To install via NPM:
+
+```
+npm install --save jsonld-document-loader
+```
+
+### Development
+
+To install locally (for development):
+
+```
+git clone https://github.com/digitalbazaar/jsonld-document-loader.git
+cd jsonld-document-loader
+npm install
+```
+
+## Usage
+
+```js
+import {JsonLdDocumentLoader} from 'jsonld-document-loader';
+// or
+const {JsonLdDocumentLoader} = require('jsonld-document-loader');
+
+const loader = new JsonLdDocumentLoader();
+```
+
+### `addStatic()`
+
+The `addStatic()` method allows developers to load fixed static contexts and
+documents, to ensure known versions and contents, without going out to the
+network.
+
+For example, to add support for the DID Core context, the VC context, and crypto 
+suite specific contexts:
+
+```js
+import cred from 'credentials-context';
+import didContext from 'did-context';
+import ed25519Ctx from 'ed25519-signature-2020-context';
+
+const {contexts: credentialsContexts, constants: {CREDENTIALS_CONTEXT_V1_URL}} =
+  cred;
+
+const jdl = new JsonLdDocumentLoader();
+
+jdl.addStatic(ed25519Ctx.CONTEXT_URL, ed25519Ctx.CONTEXT);
+
+jdl.addStatic(didContext.constants.DID_CONTEXT_URL,
+  didContext.contexts.get(didContext.constants.DID_CONTEXT_URL));
+
+jdl.addStatic(CREDENTIALS_CONTEXT_V1_URL,
+  credentialsContexts.get(CREDENTIALS_CONTEXT_V1_URL));
+
+const documentLoader = jdl.documentLoader.bind(jdl);
+// Pass to jsonld, jsonld-signatures, vc-js and similar libraries
+```
+
+### `addResolver()`
+
+To add support for resolving DIDs and DID-related key ids:
```

Comment:
This is just a specific use case, right? Can this be updated to say it can be used to combine in any other document loaders *such as* DID resolvers for DIDs and DID URLs for information (e.g., verification methods) that is expressed in DID Documents?

Reply:
Yes, this is /just/ specific to DID drivers. I was thinking of adding a separate method, like `compose()` or `addDocumentLoader()`, that allows users to combine with other document loaders.
(But did drivers and document loaders shouldn't be combined in one method, as they have very different return signatures)